Gems/Archive/Code/Source/Tools/ArchiveWriter.cpp

```diff
@@ -424,7 +431,12 @@ namespace Archive
             return false;
         }
 
-        return ReadArchiveHeaderAndToc();
+        if (!ReadArchiveHeaderAndToc())
+        {
+            UnmountArchive();
```

Comment:
might be worth a comment saying why we're calling Unmount, which it looks like is to make sure to clean up/reset any stream settings that may have been changed when reading the header and Toc

Reply:
Will do.